.github/workflows/markdown-link-check.yml

```diff
@@ -7,6 +7,12 @@ on:
     paths:
       - "**/*.md"
 
+  pull_request:
+    branches:
+      - main
```

Comment:
oh - we should chat about this. we had this here but it doesn't work when adding new addons or new pages to docs since its checking whats currently live and not what "would be live" so it fails PRs. 

not sure what the correct route is - maybe we want that and its up to reviewers to know if thats a true failure or a false failure since its getting added in the PR

Reply:
created https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/725 for testing purposes, looks like the PR check fails as expected but this is coming from a branch created from the same repo, need to test this against a PR coming from external contributor from fork